packages/fetchai/connections/soef/connection.py

```diff
@@ -251,9 +252,13 @@ def __init__(
         self.base_url = "http://{}:{}".format(soef_addr, soef_port)
         self.oef_search_dialogues = OefSearchDialogues()
 
+        data_dir = data_dir or os.getcwd()
         self._token_storage_path = token_storage_path
         if self._token_storage_path is not None:
-            self._token_storage_path = os.path.abspath(self._token_storage_path)
+            if not Path(self._token_storage_path).is_absolute():
```

Comment:
Why the check for absolute?

Reply:
because if it is absolute it doesn't make sense to concatenate the paths